doc/examples/setup_assistant/setup_assistant_tutorial.rst

```diff
@@ -52,56 +37,62 @@ Step 1: Start
-be disabled. The default value is 10,000 collision checks. Collision
-checking is done in parallel to decrease processing time.
-
-* Click on the *Self-Collisions* pane selector on the left-hand side
-  and click on the *Generate Collision Matrix* button. The
-  Setup Assistant will work for a few second before presenting you the
-  results of its computation in the main table.
-
-|before| → |after|
-
-.. |before| image:: setup_assistant_panda_self_collisions.png
-   :width: 500px
-   :align: middle
-.. |after| image:: setup_assistant_panda_self_collisions_done.png
-   :width: 500px
-   :align: middle
+The default Self-Collision matrix generator can help reduce motion planning processing time
+by disabling collision checking for pairs of links on the robot that are known to be safe.
+This is achieved by determining which pairs of links are always in collision, never in collision,
+in collision in the robot's default position, or adjacent to each other on the kinematic chain.
+
+You can set the sampling density, which determines how many random robot positions are checked for self-collision.
+Higher densities require more computation time, while lower densities may result in disabling pairs that should not be disabled.
+By default, the generator checks 10,000 random positions for self-collision.
+
+To speed up collision checking, the processing is done in parallel.
+This reduces the overall processing time for generating the collision matrix.
+
+**To generate the self-collision matrix**, first select the Self-Collisions pane on the left-hand side of the MoveIt Setup Assistant.
+Here, you can adjust the self-collision sampling density to balance between computation time and accuracy of the results.
+
+Once you have selected the desired sampling density, click on the Generate Collision Matrix button to initiate the computation.
+The Setup Assistant will take a few seconds to compute the self-collision matrix, which involves checking for pairs of links that can be safely disabled from collision checking.
+
+.. image:: collision_matrix/setup_assistant_panda_collision_matrix.png
+   :width: 700px
+   :align: center
+
+Once the computation is complete, the results will be presented in the main table. The table shows the pairs of links that have been identified as either safe or unsafe to disable from collision checking.
+Links that are safe to disable are marked with a checkmark. You can manually adjust the checkmarks as needed to enable or disable self-collision checking for specific link pairs
+
+.. image:: collision_matrix/setup_assistant_panda_collision_matrix_done.png
+   :width: 700px
+   :align: center
 
 Step 3: Add Virtual Joints
 --------------------------
+Virtual joints are used primarily to attach the robot to the world.
+For the Panda, we will define only one virtual joint attaching the *panda_link0*
+of the Panda to the *world* world frame. This virtual joint represents the motion of the base of the robot in a plane.
```

Comment:
I am truly thankful for yout comment, @MarqRazz!

My statement is wrong. It should be a fixed virtual joint, and it is optional because the Panda arm has a fixed base. I have also used single and multiple arms URDFs without virtual joints and the models worked as expected.

I did some searches and checked older documentations. My conclusion is as follows.

1. A fixed virtual joint provides fixes a robot base to the world. Here is a relevant [documentation](http://docs.ros.org/en/kinetic/api/moveit_tutorials/html/doc/urdf_srdf/urdf_srdf_tutorial.html#virtual-joints).

2. A planar virtual joint provides 3 DOF and attaches a mobile base to the world (this would be the one to use for mobile manipulators). Here is relevant [documentation](http://docs.ros.org/en/indigo/api/moveit_tutorials/html/doc/setup_assistant/setup_assistant_tutorial.html#step-3-add-virtual-joints) and a screenshot from [Mobile Manipulation workshop by PickNik team members](https://docs.google.com/presentation/d/1Qx66OICZrTRzJ73qj3tsJS0eOS2Cqw5e-JaNmpRGywQ/edit#slide=id.gf940579c23_3_6), page 13. 

![Planar joint for Stretch mobile manipulator](https://i.imgur.com/IFPiTsT.png)

3. A floating virtual joint provides 6 DOF and would be the one to use if we attach an arm to a quad-rotor for example. Here is a relevant [issue](https://github.com/ros-planning/moveit/issues/2111). 

I think the moveit_config in the resources package should use a fixed joint. I updated the tutorial based on this information. 

@sea-bass, can you please share with us how MoveIt uses the virtual joint and its advantages/disadvantages?